src/glue/v2/communication.cpp

```diff
@@ -131,59 +172,123 @@ storage::v3::Result<Value> ToBoltValue(const query::v2::TypedValue &value, const
-  if (maybe_properties.HasError()) return maybe_properties.GetError();
-  std::map<std::string, Value> properties;
-  for (const auto &prop : *maybe_properties) {
-    properties[db.PropertyToName(prop.first)] = ToBoltValue(prop.second);
+Value ToBoltValue(msgs::Value value) {
+  switch (value.type) {
+    case msgs::Value::Type::Null:
+      return {};
+    case msgs::Value::Type::Bool:
+      return {value.bool_v};
+    case msgs::Value::Type::Int64:
+      return {value.int_v};
+    case msgs::Value::Type::Double:
+      return {value.double_v};
+    case msgs::Value::Type::String:
+      return {std::string(value.string_v)};
+    case msgs::Value::Type::List: {
+      std::vector<Value> values;
+      values.reserve(value.list_v.size());
+      for (const auto &v : value.list_v) {
+        auto maybe_value = ToBoltValue(v);
+        values.emplace_back(std::move(maybe_value));
+      }
+      return Value{std::move(values)};
+    }
+    case msgs::Value::Type::Map: {
+      std::map<std::string, Value> map;
+      for (const auto &kv : value.map_v) {
+        auto maybe_value = ToBoltValue(kv.second);
+        map.emplace(kv.first, std::move(maybe_value));
+      }
+      return Value{std::move(map)};
+    }
+    case msgs::Value::Type::Vertex:
+    case msgs::Value::Type::Edge:
+    case msgs::Value::Type::Path: {
+      throw utils::BasicException("Path not supported!");
+      // auto maybe_path = ToBoltPath(value.path_v, shard_map, view);
+      // if (maybe_path.HasError()) return maybe_path.GetError();
+      // return Value(std::move(*maybe_path));
+    }
+      // TODO Value to Date types not supported
+      // case msgs::Value::Type::Date:
+      //   return Value(value.);
+      // case msgs::Value::Type::LocalTime:
+      //   return Value(value.ValueLocalTime());
+      // case msgs::Value::Type::LocalDateTime:
+      //   return Value(value.ValueLocalDateTime());
+      // case msgs::Value::Type::Duration:
+      //   return Value(value.ValueDuration());
   }
-  return communication::bolt::Vertex{id, labels, properties};
 }
 
-storage::v3::Result<communication::bolt::Edge> ToBoltEdge(const storage::v3::EdgeAccessor &edge,
-                                                          const storage::v3::Shard &db, storage::v3::View view) {
-  // TODO(jbajic) Fix bolt communication
-  auto id = communication::bolt::Id::FromUint(0);
-  auto from = communication::bolt::Id::FromUint(0);
-  auto to = communication::bolt::Id::FromUint(0);
-  const auto &type = db.EdgeTypeToName(edge.EdgeType());
-  auto maybe_properties = edge.Properties(view);
-  if (maybe_properties.HasError()) return maybe_properties.GetError();
-  std::map<std::string, Value> properties;
-  for (const auto &prop : *maybe_properties) {
-    properties[db.PropertyToName(prop.first)] = ToBoltValue(prop.second);
+Value ToBoltValue(msgs::Value value, const coordinator::ShardMap & /*shard_map*/, storage::v3::View /*view*/) {
```

Comment:
What is the difference between this and the function above?

Reply:
I will remove one